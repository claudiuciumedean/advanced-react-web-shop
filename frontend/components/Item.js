import React, { Component } from "react";
import PropTypes from "prop-types";
import Link from "next/link";

import formatMoney from "../lib/formatMoney"; 

import DeleteItem from "./DeleteItem";
import AddToCart from "./AddToCart";

import Title from "./styles/Title";
import ItemStyles from "./styles/ItemStyles";
import PriceTag from "./styles/PriceTag";

export default class Item extends Component {
    render() {
        const { id, image, title, price } = this.props.item;

        return (
            <ItemStyles>
                {image && <img src={image} alt={title}></img>}
                <Title>
                    <Link href={{ pathname: "/item", query: { id: id } }}>
                        <a>{title}</a>
                    </Link>                
                </Title>
                <PriceTag>
                    {formatMoney(price)}
                </PriceTag>
                <div className="buttonList">
                    <Link href={{
                        pathname: "update",
                        query: { id: id }
                    }}>
                        <a>Edit </a>
                    </Link>
                    <AddToCart id={id}/>
                    <DeleteItem id={id}>Delete Item</DeleteItem>
                </div>
            </ItemStyles>
        );
    }

    static propTypes = {
        item: PropTypes.object.isRequired
    };
}