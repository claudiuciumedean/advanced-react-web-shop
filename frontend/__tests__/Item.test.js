import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import Item from "../components/Item";

const sampleItem = {
    id: "123fdsfsa",
    title: "Item 1",
    price: 4000,
    description: "This item is really cool",
    image: "dog.jpg",
    largeImage: "dog_big.jpg"
};

describe("<Item/>", () => {
    it("renders and matches the snapshot", () => {
        const wrapper = shallow(<Item item={sampleItem}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});