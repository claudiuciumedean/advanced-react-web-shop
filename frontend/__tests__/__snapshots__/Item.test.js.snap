// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item/> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="Item 1"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "123fdsfsa",
          },
        }
      }
    >
      <a>
        Item 1
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "123fdsfsa",
          },
        }
      }
    >
      <a>
        Edit 
      </a>
    </Link>
    <AddToCart
      id="123fdsfsa"
    />
    <DeleteItem
      id="123fdsfsa"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
